File: frontend/src/app/core/services/servico.service.ts
import { Injectable } from '@angular/core';
import { HttpClient, HttpParams } from '@angular/common/http';
import { Observable } from 'rxjs';
import { environment } from '../../../environments/environment';

export interface Servico {
  id?: string;
  nome?: string;
  descricao?: string;
  codigoInterno?: string;
  preco?: number;
  status?: string;
  codigoServicoLc116?: string;
  aliquotaIss?: number;
  // Campos Reforma Tributária
  cnaeCodigo?: string;
  nbsCodigo?: string;
  descricaoNota?: string;
  aliquotaIbsPadrao?: number;
  aliquotaCbsPadrao?: number;
  // Audit
  criadoEm?: string;
  atualizadoEm?: string;
}

export interface Page<T> {
  content: T[];
  totalElements: number;
  totalPages: number;
  size: number;
  number: number;
}

@Injectable({
  providedIn: 'root'
})
export class ServicoService {
  private apiUrl = `${environment.apiUrl}/api/v1/servicos`;

  constructor(private http: HttpClient) { }

  findAllPaged(page: number, size: number, busca: string = ''): Observable<Page<Servico>> {
    let params = new HttpParams()
      .set('page', page.toString())
      .set('size', size.toString());

    if (busca) {
      params = params.set('busca', busca);
    }

    return this.http.get<Page<Servico>>(this.apiUrl, { params });
  }

  findById(id: string): Observable<Servico> {
    return this.http.get<Servico>(`${this.apiUrl}/${id}`);
  }

  insert(data: Servico): Observable<Servico> {
    return this.http.post<Servico>(this.apiUrl, data);
  }

  update(id: string, data: Servico): Observable<Servico> {
    return this.http.put<Servico>(`${this.apiUrl}/${id}`, data);
  }

  delete(id: string): Observable<void> {
    return this.http.delete<void>(`${this.apiUrl}/${id}`);
  }
}

File: backend/src/main/java/com/erp/capitalerp/application/servicos/ServicoService.java
package com.erp.capitalerp.application.servicos;

import com.erp.capitalerp.application.servicos.dto.ServicoDTO;
import com.erp.capitalerp.domain.servico.Servico;
import com.erp.capitalerp.infrastructure.persistence.servicos.ServicoRepository;
import jakarta.persistence.EntityNotFoundException;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

import java.util.UUID;

@Service
@Transactional
public class ServicoService {

    private final ServicoRepository servicoRepository;

    public ServicoService(ServicoRepository servicoRepository) {
        this.servicoRepository = servicoRepository;
    }

    @Transactional(readOnly = true)
    public Page<ServicoDTO> listar(String busca, Pageable pageable) {
        return servicoRepository.findComFiltros(busca, pageable).map(ServicoDTO::new);
    }

    @Transactional(readOnly = true)
    public ServicoDTO buscarPorId(UUID id) {
        Servico servico = servicoRepository.findById(id)
                .orElseThrow(() -> new EntityNotFoundException("Serviço não encontrado: " + id));
        return new ServicoDTO(servico);
    }

    @Transactional
    public ServicoDTO criar(ServicoDTO dto) {
        Servico servico = new Servico();
        copyDtoToEntity(dto, servico);
        servico = servicoRepository.save(servico);
        return new ServicoDTO(servico);
    }

    @Transactional
    public ServicoDTO atualizar(UUID id, ServicoDTO dto) {
        Servico servico = servicoRepository.findById(id)
                .orElseThrow(() -> new EntityNotFoundException("Serviço não encontrado: " + id));
        copyDtoToEntity(dto, servico);
        servico = servicoRepository.save(servico);
        return new ServicoDTO(servico);
    }

    @Transactional
    public void deletar(UUID id) {
        if (!servicoRepository.existsById(id)) {
            throw new EntityNotFoundException("Serviço não encontrado: " + id);
        }
        servicoRepository.deleteById(id);
    }

    private void copyDtoToEntity(ServicoDTO dto, Servico entidade) {
        entidade.setNome(dto.nome());
        entidade.setDescricao(dto.descricao());
        entidade.setCodigoInterno(dto.codigoInterno());
        entidade.setPreco(dto.preco());
        entidade.setStatus(dto.status());
        entidade.setCodigoServicoLc116(dto.codigoServicoLc116());
        entidade.setAliquotaIss(dto.aliquotaIss());

        // --- Campos Reforma Tributária ---
        entidade.setCnaeCodigo(dto.cnaeCodigo());
        entidade.setNbsCodigo(dto.nbsCodigo());
        entidade.setDescricaoNota(dto.descricaoNota());
        entidade.setAliquotaIbsPadrao(dto.aliquotaIbsPadrao());
        entidade.setAliquotaCbsPadrao(dto.aliquotaCbsPadrao());
    }
}
